### PR TITLE
Code refacor

### PR DESCRIPTION
**Pull request by luviiit**
Previously the code was not using the `info` object to check the added nodes instead was looping over all the `effective children` whenever there was change in nodes.

### CODE REVIEW

File: gn-poly.html [hunk 1]
```diff
@@ -81,7 +81,10 @@
       attached: function () {
         var self = this;
         this._observer = Polymer.dom(this).observeNodes( function(info) {
-          self.queryAllEffectiveChildren('gn-point').forEach(function(point) {
+          nodesToAdd = info.addedNodes.filter(function(node){
+            return node.nodeName == 'GN-POINT'; // get rid of #text nodes
```
Comment (fabianbormann) [inline gn-poly.html]: Does this work ? Means is it case insensitive?

File: gn-poly.html [hunk 2]
```diff
@@ -81,7 +81,10 @@
       attached: function () {
         var self = this;
         this._observer = Polymer.dom(this).observeNodes( function(info) {
-          self.queryAllEffectiveChildren('gn-point').forEach(function(point) {
+          nodesToAdd = info.addedNodes.filter(function(node){
```
Comment (fabianbormann) [inline gn-poly.html]: Why is this needed? Because is it not the same (attached function)

Comment (luviiit) [inline gn-poly.html]: Whenever anything is added/removed from insertion points i.e. `<content>` `queryAllEffectiveChildren('gn-point')` would take **all** the gn-point nodes and push them in the `points` array while in new code the nodes which are newly added would only be pushed to the `points` array.
Since in our code content is added only once, initially, there won't be any problem with `queryAllEffectiveChildren`. But it would be a problem if at later point of time we have to add or remove something from content. Even if something is removed from the content, `queryAllEffectiveChildren` would lead to pushing of duplicate `gn-point` nodes in the `points` array.
Did I make it clear?

File: gn-poly.html (hunk 1 above)
Comment (luviiit) [inline gn-poly.html]: I opened the console and checked for this `.nodeName` value. So I guess it is. :smiley:

**Comment by fabianbormann** [comment]
Ah ok! Thank you for make this point clear :blush:
